calico_kubernetes/calico_kubernetes_cni.py

```diff
@@ -227,73 +252,93 @@ def _set_profile_on_endpoint(endpoint, profile_name):
     """
     _log.info('Configuring Pod Profile: %s' % profile_name)
 
-    if not datastore_client.profile_exists(profile_name):
+    if not _datastore_client.profile_exists(profile_name):
         _log.info("Creating new profile %s." % (profile_name))
-        datastore_client.create_profile(profile_name)
-        # _assign_default_rules(profile_name)
+        rules = _generate_rules(container_id, namespace, pod_name, profile_name)
+        try:
+            _datastore_client.create_profile(profile_name, rules)
+        except:
+            _log.exception("Cannot create profile.")
+            sys.exit(1)
+        inbound_rules, outbound_rules = _get_annotations_rules(namespace, pod_name)
+        _apply_rules(profile_name, inbound_rules, outbound_rules)
+        _apply_tags(container_id, namespace, pod_name, profile_name)
 
     # Also set the profile for the workload.
-    datastore_client.set_profiles_on_endpoint(profile_names=[profile_name],
-                                              endpoint_id=endpoint.endpoint_id)
+    _datastore_client.set_profiles_on_endpoint(profile_names=[profile_name],
+                                               endpoint_id=endpoint.endpoint_id)
 
 
-def _assign_default_rules(profile_name):
+def _assign_ip_address():
     """
-    Generate a new profile rule list and update the datastore_client
-    :param profile_name: The profile to update
-    :type profile_name: string
+    Assigns and returns an IPv4 address using the IPAM plugin specified in CONFIG.
     :return:
     """
+    # May throw CalledProcessError - let it.  We may want to replace this with our own Exception.
```

Comment:
I'd prefer we re-raise a custom exception; good practice to encapsulate error handling as close as possible to the code that's producing it. Any callers of _assign_ip_address shouldn't need to know (or care) that it throws a CalledProcessError.